### PR TITLE
feat: UI v3 post list

### LINKED ISSUE #203: Feature: New PostList cards
### Type of feature

🍕 Feature

### Current behavior

We are combing the list and grid views. Currently we have a list card view that lacks information and hides the vote button amongst data. We need to update this.

### Suggested solution

1. Remove the list/grid toggle and place it with a list view only. 
2. This change will support #202. The hot card section will be placed above the list view.

Screen Shot 2022-07-15 at 6 37 09 AM

Link to [figma](https://www.figma.com/file/Rrc4Bc5lFoyDOrvD00ThkB/bdougie---new-hot.opensauced?node-id=0%3A1)

### Additional context

password to figma is secretsauce

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### LINKED ISSUE #204: Feature: Contributor lists hover effect.
### Type of feature

🍕 Feature

### Current behavior

On the changes in #189 we now include 5 contributors with a link to see more. Instead we will remove the link and create a hover effect to show them.

### Suggested solution

No hover
Screen Shot 2022-07-15 at 6 37 49 AM

With Hover
Screen Shot 2022-07-15 at 6 52 04 AM

To see this live, visit [app.opensauced.pizza/repos/vercel/next.js](https://app.opensauced.pizza/repos/vercel/next.js) and hove over a issue with multiple contributors. 

Link to [figma](https://www.figma.com/file/Rrc4Bc5lFoyDOrvD00ThkB/bdougie---new-hot.opensauced?node-id=0%3A1)

### Additional context

password to figma is secretsauce

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### LINKED ISSUE #261: Bug: hot repositories section avatars are too big
### Describe the bug

Make the [Avatar component](https://github.com/open-sauced/hot/blob/main/src/components/Avatar.tsx) resizable so it can take the initial 5px size described in the design:

Screenshot 2022-07-29 at 16 14 26

### Steps to reproduce

1. Go to https://hot.opensauced.pizza
2. The avatars should look like this:

Screenshot 2022-07-29 at 16 16 07

### Affected services

hot.opensauced.pizza

### Platforms

_No response_

### Browsers

_No response_

### Environment

Production

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by shamimbinnur**
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description

This PR introduces the new and updated PostList component. The updates include a new vote button and contributor hover effect. This PR also includes a new lib function for separating camelCased strings in order to leverage those titles as titles. A test was added.

## Related Tickets & Documents
Fixes #203
Fixes #261
Fixes #204

## Mobile & Desktop Screenshots/Recordings

Screen Shot 2022-07-30 at 5 42 21 AM

Screen Shot 2022-07-30 at 5 42 29 AM

## Added tests?

- [x] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

## [optional] Are there any post-deployment tasks we need to perform?

These issues needs to be addressed

https://github.com/open-sauced/hot/issues/266
https://github.com/open-sauced/hot/issues/267

### CODE REVIEW

**Comment by shamimbinnur** [comment]
Not ready yet. Opening this PR to make sure no one is gonna work on top of this.

**Comment by 0-vortex** [comment]
This PR is much cleaner and easy to test. I see some minor visual issues on mobile:
image
image

Tailwind is a mobile first framework so the above could easily be fixed for mobile in the way the hamburger menu is doing in the primary nav component.

Only other question I have is that I see the grid display being removed entirely - not sure whether that is intended?

Thanks for contributing, looking forward to get this merged in! 🍕

**Comment by 0-vortex** [comment]
> Not ready yet. Opening this PR to make sure no one is gonna work on top of this.

Ah sorry about that, started reviewing early to make sure you have the best experience! 🍕 

It looks pretty good already 😁

**Comment by shamimbinnur** [comment]
> This PR is much cleaner and easy to test. I see some minor visual issues on mobile: image image
> 
> Tailwind is a mobile first framework so the above could easily be fixed for mobile in the way the hamburger menu is doing in the primary nav component.
> 
> Only other question I have is that I see the grid display being removed entirely - not sure whether that is intended?
> 
> Thanks for contributing, looking forward to get this merged in! 🍕

@0-vortex grid display is removed as per @bdougie 's direction on the issue. But the component still exists in the directory. We can reuse it if needed.

**Comment by bdougie** [comment]
> @0-vortex grid display is removed as per @bdougie 's direction on the issue. But the component still exists in the directory. We can reuse it if needed.

We will want to add this to the future design system.

**Comment by 0-vortex** [comment]
One change I would like to request based on the design in figma is: https://tailwindcss.com/docs/font-size

We cannot have mobile friendly design if we provide absolute pixel values, however we can adjust classes based on screen size. In order to have full flexibility of the viewport setting fonts in relative classes (base, xs, lg, etc) is better and more intuitive, while we can still use absolute values when needed. On zoomed clients and screen readers this change will provide better accessibility too 🍕

**Comment by shamimbinnur** [comment]
We don't have any design for the Nav. Any suggestion for that? @bdougie

**Comment by shamimbinnur** [comment]
I have used the existing function for the voting system here. Confused, does that work? @0-vortex

**Comment by 0-vortex** [comment]
> I have used the existing function for the voting system here. Confused, does that work? @0-vortex

Existing vote function should work on list repos, not the hot 3 repositories!

**Comment by shamimbinnur** [comment]
> > I have used the existing function for the voting system here. Confused, does that work? @0-vortex
> 
> Existing vote function should work on list repos, not the hot 3 repositories!

@0-vortex  Can you please double-check? It seems voting doesn't work even in the main branch :((

**Comment by 0-vortex** [comment]
Main branch works just fine in this section if logged in:

Screenshot 2022-07-28 at 13 59 27

**Comment by bdougie** [comment]
@open-sauced/triage check it out

File: src/components/Avatar.tsx
Comment (0-vortex) [inline src/components/Avatar.tsx:42]: this is just formatting

File: src/components/Avatar.tsx
Comment (0-vortex) [inline src/components/Avatar.tsx:48]: same here

File: tailwind.config.cjs
Comment (0-vortex) [inline tailwind.config.cjs:29]: where are these new variables used and how was their naming decided?

**Review by 0-vortex — CHANGES_REQUESTED** [review]
Minor chores, testing visually from mobile now 🍕

**Comment by 0-vortex** [comment]
Load more button is displaced still:

File: tailwind.config.cjs
Comment (shamimbinnur) [reply tailwind.config.cjs:29]: Those variables are used to create a gradient color for the header on the hero section. Whereas 'gradFirst' is the start, and 'gradLast' is the ending color of the gradient.

Comment (0-vortex) [reply tailwind.config.cjs:29]: #230 ?

**Comment by shamimbinnur** [comment]
> https://deploy-preview-242--hot-sauced-ui.netlify.app/

That's white already. You're not seeing the latest preview somehow.
Screenshot 2022-07-29 at 8 57 49 PM

File: tailwind.config.cjs
Comment (shamimbinnur) [reply tailwind.config.cjs:29]: Nope, sir! That's for the header text! 
Screenshot 2022-07-29 at 9 01 04 PM

**Comment by 0-vortex** [comment]
> > https://deploy-preview-242--hot-sauced-ui.netlify.app/
> 
> 
> 
> That's white already. You're not seeing the latest preview somehow.
> 
> Screenshot 2022-07-29 at 8 57 49 PM
> 
> 
> 
> 

talking about positioning here, not color, it should be positioned in the middle of the screen not the left 🍕

**Comment by shamimbinnur** [comment]
> Load more button is displaced still:
> 
> 

> > > https://deploy-preview-242--hot-sauced-ui.netlify.app/
> > 
> > 
> > That's white already. You're not seeing the latest preview somehow.
> > Screenshot 2022-07-29 at 8 57 49 PM
> 
> talking about positioning here, not color, it should be positioned in the middle of the screen not the left 🍕

@0-vortex  Middle for both mobile and desktop? With the same wide?

File: tailwind.config.cjs
Comment (bdougie) [reply tailwind.config.cjs:29]: FWIW, the grad shortening is not entirely clear until you spelled it out. I think I will open an issue to update that to state "gradientFirst" etc.

Not blocking for this, but we should address that.

File: src/components/ListDisplay.tsx
Comment (bdougie) [inline src/components/ListDisplay.tsx:19]:
```suggestion
        <h1 className='text-[24px] text-white font-semibold'>${activelink} repositories</h1>
```

We should rename the title of the list based on the filter here.

File: src/components/ListDisplay.tsx
Comment (bdougie) [inline src/components/ListDisplay.tsx:14]: I'd prefer the name `ListRepositories` instead of `ListDisplay`

**Review by bdougie — CHANGES_REQUESTED** [review]
Just 2 changes I saw, but could also expand your description to explain how you implement the hover effect for contributors.

File: src/components/PostList.tsx
Comment (bdougie) [inline src/components/PostList.tsx:75]:
```suggestion
                        <div className='w-[24px] h-[24px] overflow-hidden rounded-full -mr-[15px] transition-all duration-300 hover:mr-0 hover:scale-150'>
```

Switch up the hover:scale to 150

File: src/components/ListDisplay.tsx
Comment (0-vortex) [inline src/components/ListDisplay.tsx:18]: I have mentioned we cannot use `img`s as icon placeholders before, but if we do, they have to conform with our previous patterns and have correct markup and a11y. You were the first to introduce the bg-url pattern in tailwind so either re-use that or add better alt text. 🍕

File: src/components/PostList.tsx
Comment (0-vortex) [inline src/components/PostList.tsx:70]: Please use the correct tailwind classes for text size: https://tailwindcss.com/docs/font-size

File: src/components/PostList.tsx
Comment (0-vortex) [inline src/components/PostList.tsx:87]: you cannot put a paragraph inside a button, please replace this with any inline item 🍕

File: tailwind.config.cjs
Comment (0-vortex) [reply tailwind.config.cjs:29]: Making the gradient text classes more intuitive would be nice:
- `headerTextGradientStart/headerTextGradientMiddle/headerTextGradientEnd` - 👎 
- `headerTextGradient0/headerTextGradient50/headerTextGradient100` - 👍 

However, this is the wrong approach as we do not have the colors available as (actually) re-usable classes, we should create the gradient based on existing theme colors, rather than create duplicate color classes subsets for the gradients. 

Visually, our start, middle and end color colors are not perceived by our users, it is unlikely the end and start color stops need additional hex colors other than the ones we have. I'm keen to stop introducing unwanted things we then have to open new fix PRs for.

**Review by 0-vortex — CHANGES_REQUESTED** [review]
I'm seeing lots of things we have discussed in the previous PRs

Other than the code comments, replace the SVGs with the following icons please:
- `issueIconGrey.svg`: https://react-icons.github.io/react-icons/search?q=issue
- `popularIcon.svg`: https://react-icons.github.io/react-icons/search?q=calendar
- `pullIconGrey`: https://react-icons.github.io/react-icons/search?q=pull
- `starIconGrey`: https://react-icons.github.io/react-icons/search?q=star

**Comment by 0-vortex** [comment]
Some examples of icons being used in code:
Screenshot 2022-07-30 at 08 37 28

Please re-use those bits of code instead of adding images

File: tailwind.config.cjs
Comment (bdougie) [reply tailwind.config.cjs:29]: open this issue https://github.com/open-sauced/hot/issues/266

That change should happen after this PR.

**Comment by bdougie** [comment]
@0-vortex I will wait for your review before merging.

**Review by 0-vortex — APPROVED** [review]

**Comment by jsmith2320** [comment]
Changes look great

**Pull request merged**
